api/src/main/java/com/javadiscord/jdi/core/api/utils/CacheUpdater.java

```diff
@@ -0,0 +1,44 @@
+package com.javadiscord.jdi.core.api.utils;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+import com.javadiscord.jdi.internal.cache.Cache;
+
+public class CacheUpdater {
+
+    private final Cache cache;
+
+    public CacheUpdater(Cache cache) {
+        this.cache = cache;
+    }
+
+    public <T> void updateCache(T result) {
+        if (result == null) {
+            return;
+        }
+        try {
+            Field guildIdField = result.getClass().getDeclaredField("guildId");
+            Field idField = result.getClass().getDeclaredField("id");
+
+            long guildId = getLongFromField(guildIdField, result);
+            long id = getLongFromField(idField, result);
+
+            cache.getCacheForGuild(guildId).add(id, result);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            // NoSuchFieldException: don't cache result
```

Comment:
Could you add a `trace` log message just so if this occurs we can at least know about it for potential bug fixing in the future please?

Reply:
Added log tracing for this.

Reply:
Almost there, instead of `LOGGER.log(Level.TRACE` you can simply do `LOGGER.trace()`

Reply:
Updated to use LOGGER.trace